CMakeLists.txt

```diff
@@ -86,13 +96,25 @@ else ()
 
     # -- Dependencies --
     add_subdirectory(dependencies)
+    if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+        # Testing Dependencies
+        if (BUILD_TESTING)
+            message(STATUS "Building tests")
+            # Testing dependency
+            find_package(GTest 1.11 REQUIRED)
```

Comment:
why not follow HEAD? from [their page](https://github.com/google/googletest)

```
GoogleTest now follows the Abseil Live at Head philosophy. We recommend [updating to the latest commit in the main branch as often as possible.
```

Reply:
Because we know what version works with our build at this particular version. If head is not backward compatible at some point tests of old versions of SDK will not work anymore.